invenio_rdm_records/resources/serializers/codemeta/schema.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Codemeta schema."""
+
+from edtf import parse_edtf
+from edtf.parser.grammar import ParseException
+from marshmallow import ValidationError, fields, missing
+from pydash import py_
+
+from ..schemaorg.schema import SchemaorgSchema
+
+
+class CodemetaSchema(SchemaorgSchema):
+    """CodeMeta schema.
+
+    CodeMeta schema derives from Schema.org
+    """
+
+    context = fields.Constant(
+        "https://doi.org/10.5063/schema/codemeta-2.0", data_key="@context"
+    )
+    funding = fields.Method("get_funding")
+    embargoDate = fields.Method("get_embargo_date", data_key="embargoDate")
+
+    def get_type(self, obj):
+        """Get type.
+
+        For CodeMeta, only "SoftwareSourceCode" types are allowed.
+        """
+        read_type = super().get_type(obj)
+        if "SoftwareSourceCode" not in read_type:
+            raise ValidationError(
+                "Record is not CodeMeta serializable. It must be of type software."
+            )
+        return read_type
+
+    def get_funding(self, obj):
```

Comment:
minor/shelve: we could move this up in the parent schema

Reply:
I added it here since codemeta schema re-defines funding: 

https://raw.githubusercontent.com/codemeta/codemeta/2.0/codemeta.jsonld

Now that you mention it, their [docs](https://codemeta.github.io/terms/) say the type of `funding` is  "Text"  and describe it as " Funding source (e.g. specific grant)". So we should return a string here, wdyt?

Reply:
This has been an incredibly long-running issue....hopefully it will be fixed at some point https://github.com/codemeta/codemeta/pull/274